drivers/gpio/gpio_rpi_pico.c

```diff
@@ -119,6 +117,8 @@ static int gpio_rpi_pin_interrupt_configure(const struct device *dev,
 			}
 		}
 		gpio_set_irq_enabled(pin, events, true);
+	} else {
+		gpio_set_irq_enabled(pin, ALL_EVENTS, false);
```

Comment:
I am not familiar with the pico hardware, however, from the above code it seems that to disable an event we need to explicitly call `gpio_set_irq_enabled` with the event flag set and the `enabled` parameter equal `false`. I.e. the call to
```
	gpio_set_irq_enabled(pin, events, true);
```
is only able to enable the events. In that case the call to
```
	gpio_set_irq_enabled(pin, ALL_EVENTS, false);
```
should be placed at the function entry and executed unconditionally. Otherwise if the user calls
```
gpio_pin_interrupt_configure(dev, pin, GPIO_INT_EDGE_RISING);
gpio_pin_interrupt_configure(dev, pin, GPIO_INT_EDGE_FALLING);
```
they will configure the hardware to trigger on both edges, rather than switch from triggering on the rising to the falling edge.

Reply:
Good point. I'll try putting this early unconditionally, and test. Thanks!

Reply:
Ok, yes, fixed.